front/components/agent_builder/AgentBuilderLayout.tsx

```diff
@@ -99,17 +100,31 @@ export function AgentBuilderLayout({
                   collapsible={true}
                   onCollapse={handlePanelCollapse}
                   onExpand={handlePanelExpand}
-                  className={
+                  className={cn(
                     !isResizing
                       ? "overflow-hidden transition-all duration-300 ease-in-out"
-                      : "overflow-hidden"
-                  }
+                      : "overflow-hidden",
+                    !isPreviewPanelOpen && "hidden md:block",
+                    isPreviewPanelOpen && "md:relative",
+                    isPreviewPanelOpen &&
+                      "absolute inset-0 z-50 bg-background md:relative md:inset-auto md:z-auto"
+                  )}
                 >
                   <div className="h-full w-full overflow-y-auto">
                     {rightPanel}
                   </div>
                 </ResizablePanel>
               </ResizablePanelGroup>
+
+              {!isPreviewPanelOpen && (
+                <Button
+                  label="Preview"
+                  variant="ghost"
+                  onClick={handlePanelExpand}
+                  className="fixed bottom-3 left-4 z-40 md:hidden"
```

Comment:
We try to not use z indexes as much as possible. Can we do without here?

Reply:
Also, that looks very hacky to me, no? Why not modifying the footer component to accept an extra button?